cairo/crates/pragma_dispatcher/src/contract.cairo

```diff
@@ -0,0 +1,254 @@
+            self: @ContractState, data_type: DataType, aggregation_mode: AggregationMode,
+        ) -> PragmaPricesResponse {
+            let pragma_dispatcher = IPragmaABIDispatcher {
+                contract_address: self.pragma_oracle_address.read()
+            };
+
+            pragma_dispatcher.get_data(data_type, aggregation_mode)
+        }
+    }
+
+    impl SummaryStatsWrapper of ISummaryStatsWrapper<ContractState> {
+        /// Calls calculate_mean from the Summary Stats contract.
+        fn call_calculate_mean(
+            self: @ContractState,
+            data_type: DataType,
+            aggregation_mode: AggregationMode,
+            start_timestamp: u64,
+            end_timestamp: u64,
+        ) -> SummaryStatsComputation {
+            let summary_stats_dispatcher = ISummaryStatsABIDispatcher {
+                contract_address: self.summary_stats_address.read()
+            };
+
+            summary_stats_dispatcher
+                .calculate_mean(data_type, start_timestamp, end_timestamp, aggregation_mode)
+        }
+
+        /// Calls calculate_volatility from the Summary Stats contract.
+        fn call_calculate_volatility(
+            self: @ContractState,
+            data_type: DataType,
+            aggregation_mode: AggregationMode,
+            start_timestamp: u64,
+            end_timestamp: u64,
+            num_samples: u64,
+        ) -> SummaryStatsComputation {
+            let summary_stats_dispatcher = ISummaryStatsABIDispatcher {
+                contract_address: self.summary_stats_address.read()
+            };
+
+            summary_stats_dispatcher
+                .calculate_volatility(
+                    data_type, start_timestamp, end_timestamp, num_samples, aggregation_mode
+                )
+        }
+
+        /// Calls calculate_twap from the Summary Stats contract.
+        fn call_calculate_twap(
+            self: @ContractState,
+            data_type: DataType,
+            aggregation_mode: AggregationMode,
+            start_timestamp: u64,
+            duration: u64,
+        ) -> SummaryStatsComputation {
+            let summary_stats_dispatcher = ISummaryStatsABIDispatcher {
+                contract_address: self.summary_stats_address.read()
+            };
+
+            summary_stats_dispatcher
+                .calculate_twap(data_type, aggregation_mode, duration, start_timestamp)
+        }
+    }
+
+    // ================== PRIVATE IMPLEMENTATIONS ==================
+
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
+        fn initializer(
+            ref self: ContractState,
+            owner: ContractAddress,
+            pragma_oracle_address: ContractAddress,
+            summary_stats_address: ContractAddress,
+            pragma_feed_registry_address: ContractAddress,
+            hyperlane_mailbox_address: ContractAddress,
+        ) {
+            // [Check]
+            assert(!owner.is_zero(), 'Owner cannot be 0');
```

Comment:
We can create a mod Error and add the error there (nitpick)

Reply:
done